[export] Serialize union fields with single entry dict.

zhxchen17 opened the pull request:
Summary: remove "$type" and "$value" fields, instead only serialize as {type: value} for union fields directly.

Test Plan: CI

Differential Revision: D54553770

tugsbayasgalan commented on torch/_export/serde/schema.py:
```diff
@@ -8,7 +8,7 @@
 
 # NOTE: Please update this value if any modifications are made to the schema
-SCHEMA_VERSION = (4, 2)
+SCHEMA_VERSION = (5, 1)
```
Why does it go down from 2 to 1? Should we leave a comment explain what these numbers are for?

zhxchen17 replied:
here is a doc https://docs.google.com/document/d/1EN7JrHbOPDhbpLDtiYG4_BPUs7PttpXlbZ27FuwKhxg/edit

zhxchen17 committed:
[export] Serialize union fields with single entry dict. (#121263)

Summary:

remove "$type" and "$value" fields, instead only serialize as {type: value} for union fields directly.

Test Plan: CI

Reviewed By: tugsbayasgalan

Differential Revision: D54553770

osalpekar commented:
@pytorchbot revert -m "A large number of inductor benchmarking jobs failing starting this PR. See for details: https://hud.pytorch.org/pytorch/pytorch/commit/7feabe9b73e6ba7724b62ea91df27049defdf378" -c ghfirst
